mne/inverse_sparse/tests/test_mxne_optim.py

```diff
@@ -40,86 +43,175 @@ def test_l21_mxne():
     X[4] = -2
     M = np.dot(G, X)
 
-    X_hat_prox, active_set, _ = mixed_norm_solver(M,
-                            G, alpha, maxit=1000, tol=1e-8,
-                            active_set_size=None, debias=True,
-                            solver='prox')
+    X_hat_prox, active_set, _ = mixed_norm_solver(
+        M, G, alpha, maxit=1000, tol=1e-8, active_set_size=None,
+        debias=True, solver='prox')
     assert_array_equal(np.where(active_set)[0], [0, 4])
-    X_hat_cd, active_set, _ = mixed_norm_solver(M,
-                            G, alpha, maxit=1000, tol=1e-8,
-                            active_set_size=None, debias=True,
-                            solver='cd')
+    X_hat_cd, active_set, _ = mixed_norm_solver(
+        M, G, alpha, maxit=1000, tol=1e-8, active_set_size=None,
+        debias=True, solver='cd')
     assert_array_equal(np.where(active_set)[0], [0, 4])
-    assert_array_almost_equal(X_hat_prox, X_hat_cd, 5)
+    X_hat_bcd, active_set, _ = mixed_norm_solver(
+        M, G, alpha, maxit=1000, tol=1e-8, active_set_size=None,
+        debias=True, solver='bcd')
+    assert_array_equal(np.where(active_set)[0], [0, 4])
+    assert_allclose(X_hat_prox, X_hat_cd, rtol=1e-2)
+    assert_allclose(X_hat_prox, X_hat_bcd, rtol=1e-2)
+    assert_allclose(X_hat_bcd, X_hat_cd, rtol=1e-2)
 
-    X_hat_prox, active_set, _ = mixed_norm_solver(M,
-                            G, alpha, maxit=1000, tol=1e-8,
-                            active_set_size=2, debias=True,
-                            solver='prox')
+    X_hat_prox, active_set, _ = mixed_norm_solver(
+        M, G, alpha, maxit=1000, tol=1e-8, active_set_size=2,
+        debias=True, solver='prox')
     assert_array_equal(np.where(active_set)[0], [0, 4])
-    X_hat_cd, active_set, _ = mixed_norm_solver(M,
-                            G, alpha, maxit=1000, tol=1e-8,
-                            active_set_size=2, debias=True,
-                            solver='cd')
+    X_hat_cd, active_set, _ = mixed_norm_solver(
+        M, G, alpha, maxit=1000, tol=1e-8, active_set_size=2,
+        debias=True, solver='cd')
     assert_array_equal(np.where(active_set)[0], [0, 4])
-    assert_array_almost_equal(X_hat_prox, X_hat_cd, 5)
+    X_hat_bcd, active_set, _ = mixed_norm_solver(
+        M, G, alpha, maxit=1000, tol=1e-8, active_set_size=2,
+        debias=True, solver='bcd')
+    assert_array_equal(np.where(active_set)[0], [0, 4])
+    X_hat_gbcd, active_set, _ = mixed_norm_solver(
+        M, G, alpha, maxit=1000, tol=1e-8, active_set_size=2,
+        debias=True, solver='gbcd')
```

Comment:
gbcd was an implementation of a BCD version called greedy block coordinate descent. (Wei et al., DOA Estimation Using a Greedy Block Coordinate Descent Algorithm, IEEE Transactions on Signal Processing 60(12), 2012) I took it out of the pull request and obviously forgot to take out the test, too. You can just delete it. We can still think of adding more options for optimization later.